liferay-tango/liferay-portal
# LPS-119068 [Page Audit] Support Site configuration.

**Motivation**

To ensure that a page is optimized for search engine indexing and visibility, it is necessary to check the page's compliance with best practices in terms of performance, accessibility, search engine metadata and SEO. Page audit test allows to make a quick check of that compliance and get warnings and tips to improve each one of the categories.

As an API key for the Page Speed service is needed, we must include some information in the panel to facilitate on-boarding.

We also need to add a new icon in the Control bar to display the panel.

**What is included in this PR**

**Given** the Page Audit Panel
**When** PageSpeed's API Key is not configured and the user has permissions for accessing to the site settings
**Then** the button Connect with Page Speed leads to the Site Settings > Pages > PageSpeed.

**How to test it**

Integration tests are included in this PR.

If you want to test it manually:

1. Create a user with the Role Site Admin of the current Site
2. Log in with the new user
3. Open the Page Audit Panel
4. Click the button Connect Page Speed
5. Check that you have been redirected to Site Settings

https://user-images.githubusercontent.com/5776822/111678376-819fda00-8820-11eb-8967-069e27fe1930.mov
